### DISCUSSION TITLE
Problems encountered using raga-app

### DISCUSSION THREAD

**Discussion by gabrielajsanchez** [adidahiya/music-library-scripts#255] [General]
Hi there!

I am a DJ and stumbled onto your product, raga-app, recently and am really interested in using it music management in preparation for DJ mixing. I recently downloaded the application and have been trying to export my Swinsian Library, but I have encountered a few problems.

I raise two questions that I hope to receive support from you or any user of the application:

1. What is the best way to convert an `.m3u8` file into an `.xml` file in order to use in `raga-app`?
2. How does one go about clearing `raga-app` in order to re-upload another `.xml` file?

---

### Context

1. In Swinsian, it appears that I do not have the option to export my entire library. One potential reason is because I've set my library to live in an external hard drive rather than my local machine in order to conserve space. 

<img width="932" alt="Screenshot 2024-10-15 at 3 12 33 PM" src="https://github.com/user-attachments/assets/c6be2209-6415-4cde-9998-4a589b5e2eb5">

2. Instead of exporting my library from Swinsian, I instead opted to export a single playlist, however, Swinsian only exports playlists in an `.m3u8` format (I'm also uncertain whether this is due to my music collection being stored in an external hard drive). I tried finding a way to convert to an `.xml` format online and believed to have found an approach, however, my approach proved incorrect, and every time I now open raga-app, I receive the following output: 

<img width="1424" alt="Screenshot 2024-10-15 at 3 16 15 PM" src="https://github.com/user-attachments/assets/6f1fb3ed-b6c4-4753-a207-145baeb8ce61">

3. Even after uninstalling the application `rm -rf /Applications/raga-app.app` and re-downloading, the same error appears when opening the application.

**Comment by adidahiya** [comment]
> In Swinsian, it appears that I do not have the option to export my entire library. One potential reason is because I've set my library to live in an external hard drive rather than my local machine in order to conserve space.

I also keep my tracks on an external hard drive, but the Swinsian library itself (database and XML file) lives on my internal hard drive.

You might need to be on the beta release channel of Swinsian. You can access this by holding the option key while the 'Swinsian' menu bar dropdown is open and clicking 'Check for Updates (Beta)'. Try upgrading to the latest beta version (I'm on 3.0 preview 19):

![Screenshot 2024-10-16 at 9 04 12 AM](https://github.com/user-attachments/assets/b766f99f-5f5b-420d-9c87-2c80da52c3ba)

![Screenshot 2024-10-16 at 9 07 38 AM](https://github.com/user-attachments/assets/6bc0cf2e-7e75-4156-82f6-fcbaa63bdc35)


raga won't work for m3u playlists. It's currently only designed for Swinsian's XML library format.

Thanks for giving it a try. Happy to answer further questions if you have them.
